crates/protocol/src/types/version.rs

```diff
@@ -0,0 +1,20 @@
+//! A protocol version number backed by [`crate::ProtocolVersion`]
+
+use bytes::{Buf, BufMut};
+
+use crate::{error, version::ProtocolVersion, ProtocolType};
+
+impl ProtocolType for ProtocolVersion {
+    type DecodeError = super::varint::DecodeError;
```

Comment:
Let's import `crate::types::VarInt` and use it here (`VarInt::DecodeError`) and below.

Reply:
```
ambiguous associated type
version.rs(9, 24): use the fully-qualified path: `<varint::VarInt as ProtocolType>::DecodeError`
```

We'll end up with more ugly code than we started

Reply:
Then let's just replace `super`s with `crate`s (as in the other comments).

Reply:
How is `crate` better than `super`? It's not like the type modules are deeply nested or anything

Reply:
Discussed on Discord